lib/services/responses.js

```diff
@@ -0,0 +1,191 @@
+
+/**
+ * Handles errors in the standard/generic way
+ * @param {Object} res
+ * @returns {Function}
+ */
+function handleError(res) {
+  return ({ code, stack }) => {
+    if (code) {
+      res.status(code).send(stack);
+    } else {
+      res.send(stack);
+    }
+  };
+}
+
+/**
+ * Responds with JSON and capture
+ * Captures and hides appropriate errors.
+ * These return JSON always, because these endpoints are JSON-only.
+ * @param {Function} fn
+ * @param {Object} res
+ * @returns {Promise}
+ */
+function expectJSON(fn, res) {
+  return bluebird.try(fn).then((result) => res.json(result)).catch(handleError(res));
+}
+
+/**
+ * Creates a method validation middleware.
+ * @param {{allow: [String]}} options
+ * @returns {Function}
+ */
+function methodNotAllowed(options) {
+  const allowed = options.allow;
+
+  return (req, res, next) => {
+    let message, code,
+      method = req.method;
+
+    if (_includes(allowed, method.toLowerCase())) {
+      next();
+    } else {
+      code = 405;
+      message = `Method ${method} not allowed`;
+      res.set('Allow', allowed.join(', ').toUpperCase());
+      sendDefaultResponseForCode(code, message, res, options);
+    }
+  };
+}
+
+/**
+ * Sends whatever is default for this type of data with this status code.
+ * @param {number} code
+ * @param {Object} res
+ * @returns {function}
+ */
+function sendDefaultErrorCode(code, res) {
+  return () => {
+    res.sendStatus(code);
+  };
+}
+
+/**
+ * Sends some html (should probably be some default, or a render of a 500 page).
+ * @param {number} code
+ * @param {string} message
+ * @param {Object} res
+ * @returns {function}
+ */
+function sendHTMLErrorCode(code, message, res) {
+  return () => {
+    res.type('html');
+    res.send(`${code} ${message}`);
+  };
+}
+
+/**
+ * Sends some json error.
+ * @param {number} code
+ * @param {string} message
+ * @param {Object} res
+ * @param {Object} extras
+ * @returns {function}
+ */
+function sendJSONErrorCode(code, message, res, extras) {
+  return () => {
+    res.json(_assign({ message, code }, extras));
+  };
+}
+
+/**
+ * Sends some text error.
+ * @param {number} code
+ * @param {string} message
+ * @param {Object} res
+ * @returns {function}
+ */
+function sendTextErrorCode(code, message, res) {
+  return () => {
+    res.type('text');
+    res.send(`${code} ${message}`);
+  };
+}
+
+/**
+ * Sends default response for code.
+ * @param {number} code
+ * @param {string} message
+ * @param {Object} res
+ * @param {Object} [extras]
+ */
+function sendDefaultResponseForCode(code, message, res, extras) {
+  res.status(code).format({
+    json: sendJSONErrorCode(code, message, res, extras),
+    html: sendHTMLErrorCode(code, message, res),
+    text: sendTextErrorCode(code, message, res),
+    default: sendDefaultErrorCode(code, res)
+  });
+}
+
+/**
+ * Creates an accept validation middleware.
+ * @param {{accept: [string]}} options
+ * @returns {Function}
+ */
+function notAcceptable(options) {
```

Comment:
ditto here

Reply:
my comment above should cover this one too